File: src/jvm/package/src/doc-examples/java/questpdf/docexamples/PreventPageBreakExamples.java
package questpdf.docexamples;

import com.questpdf.fluent.Document;
import com.questpdf.helpers.Colors;
import com.questpdf.helpers.PageSizes;
import org.junit.jupiter.api.Test;

public class PreventPageBreakExamples extends DocExample {

    @Test
    public void enabledExample() {
        Document
            .create(document -> {
                document.page(page -> {
                    page.size(PageSizes.getA5());
                    page.defaultTextStyle(style -> style.fontSize(20f));
                    page.margin(30f);

                    page.content()
                        .column(column -> {
                            column.item().height(400f).background(Colors.Grey.getLighten3());
                            column.item().height(30f);

                            column.item()
                                .preventPageBreak()
                                .text(text -> {
                                    text.paragraphSpacing(15f);

                                    text.span("Optimizing Content Placement").bold().fontColor(Colors.Blue.getDarken2()).fontSize(24f);
                                    text.span("\n");
                                    text.span("By carefully determining where to place a page break, you can avoid awkward text separations and maintain readability. Thoughtful formatting improves the overall user experience, making complex topics easier to digest.");
                                });
                        });
                });
            })
            .generatePdf(output("prevent-page-break-enabled.pdf"));
    }

    @Test
    public void disabledExample() {
        Document
            .create(document -> {
                document.page(page -> {
                    page.size(PageSizes.getA5());
                    page.defaultTextStyle(style -> style.fontSize(20f));
                    page.margin(30f);

                    page.content()
                        .column(column -> {
                            column.item().height(400f).background(Colors.Grey.getLighten3());
                            column.item().height(30f);

                            column.item()
                                .text(text -> {
                                    text.paragraphSpacing(15f);

                                    text.span("Optimizing Content Placement").bold().fontColor(Colors.Blue.getDarken2()).fontSize(24f);
                                    text.span("\n");
                                    text.span("By carefully determining where to place a page break, you can avoid awkward text separations and maintain readability. Thoughtful formatting improves the overall user experience, making complex topics easier to digest.");
                                });
                        });
                });
            })
            .generatePdf(output("prevent-page-break-disabled.pdf"));
    }
}

File: src/jvm/package/src/doc-examples/kotlin/questpdf/docexamples/PreventPageBreakExamples.kt
package questpdf.docexamples

import com.questpdf.fluent.Document
import com.questpdf.helpers.Colors
import com.questpdf.helpers.PageSizes
import org.junit.jupiter.api.Test

class PreventPageBreakExamples : DocExample() {

    @Test
    fun enabledExample() {
        Document
            .create {
                page {
                    size(PageSizes.A5)
                    defaultTextStyle { fontSize(20f) }
                    margin(30f)

                    content()
                        .column {
                            item().height(400f).background(Colors.Grey.Lighten3)
                            item().height(30f)

                            item()
                                .preventPageBreak()
                                .text {
                                    paragraphSpacing(15f)

                                    span("Optimizing Content Placement").bold().fontColor(Colors.Blue.Darken2).fontSize(24f)
                                    span("\n")
                                    span("By carefully determining where to place a page break, you can avoid awkward text separations and maintain readability. Thoughtful formatting improves the overall user experience, making complex topics easier to digest.")
                                }
                        }
                }
            }
            .generatePdf(output("prevent-page-break-enabled.pdf"))
    }

    @Test
    fun disabledExample() {
        Document
            .create {
                page {
                    size(PageSizes.A5)
                    defaultTextStyle { fontSize(20f) }
                    margin(30f)

                    content()
                        .column {
                            item().height(400f).background(Colors.Grey.Lighten3)
                            item().height(30f)

                            item()
                                .text {
                                    paragraphSpacing(15f)

                                    span("Optimizing Content Placement").bold().fontColor(Colors.Blue.Darken2).fontSize(24f)
                                    span("\n")
                                    span("By carefully determining where to place a page break, you can avoid awkward text separations and maintain readability. Thoughtful formatting improves the overall user experience, making complex topics easier to digest.")
                                }
                        }
                }
            }
            .generatePdf(output("prevent-page-break-disabled.pdf"))
    }
}
